src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -533,13 +535,20 @@ class ArnoldAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 
 			for( IECore::CompoundObject::ObjectMap::const_iterator it = attributes->members().begin(), eIt = attributes->members().end(); it != eIt; ++it )
 			{
-				if( !boost::starts_with( it->first.string(), "user:" ) )
+				if( boost::starts_with( it->first.string(), "user:" ) )
 				{
-					continue;
+					if( const IECore::Data *data = IECore::runTimeCast<const IECore::Data>( it->second.get() ) )
+					{
+						m_user[it->first] = data;
+					}
 				}
-				if( const IECore::Data *data = IECore::runTimeCast<const IECore::Data>( it->second.get() ) )
+				if( boost::starts_with( it->first.string(), g_lightFilterPrefix.string() ) )
 				{
-					m_user[it->first] = data;
+					if( it->second )
```

Comment:
Any good reason to test against NULL here? As far as I'm aware, a NULL here would indicate a programming error somewhere else, in which case I'd rather find out about it than gloss over it.